invisible_cities/cities/dorothea_test.py

```diff
@@ -101,21 +103,56 @@ def test_dorothea_filter_events(config_tmpdir, Kr_pmaps_run4628_filename):
     assert np.all(dst.s2_peak.values == s2_peak_pass)
 
 
-def test_dorothea_exact_result(ICDATADIR, output_tmpdir):
-    file_in     = os.path.join(ICDATADIR    ,  "Kr83_nexus_v5_03_00_ACTIVE_7bar_3evts.PMP.h5")
-    file_out    = os.path.join(output_tmpdir,                      "exact_result_dorothea.h5")
-    true_output = os.path.join(ICDATADIR    , "Kr83_nexus_v5_03_00_ACTIVE_7bar_3evts.KDST.h5")
+@mark.parametrize("include_mc", (False, True))
+def test_dorothea_contains_all_tables(ICDATADIR, output_tmpdir,  include_mc):
+
+    label       = "with" if include_mc else "without"
+    file_out    = os.path.join(output_tmpdir, f"dorothea_KDST_{label}_MC.h5")
+    file_in     = os.path.join(ICDATADIR, "Kr83_nexus_v5_03_00_ACTIVE_7bar_3evts.PMP.h5")
+
+    conf = configure('dorothea invisible_cities/config/dorothea.conf'.split())
+    conf.update(dict(run_number  = -6340, 
+                     files_in    = file_in,
+                     file_out    = file_out,
+                     event_range = all_events,
+                     include_mc  = include_mc))
+
+    dorothea(**conf)
 
+    with tb.open_file(file_out) as h5out:
+        assert "Run"                  in h5out.root
+        assert "Run/events"           in h5out.root
+        assert "Run/runInfo"          in h5out.root
+        assert "Filters"              in h5out.root
+        assert "Filters/s12_selector" in h5out.root
+        assert "DST"                  in h5out.root
+        assert "DST/Events"           in h5out.root
+        if include_mc:
+            assert "MC"               in h5out.root
+            assert "MC/hits"          in h5out.root
+            assert "MC/particles"     in h5out.root
```

Comment:
Add also an `else` clause to check that these tables don't exist if `include_mc` is False.
Another (fancier) way would be to do
`assert not (node_name in h5out.root) ^ include_mc` 
which covers both cases without the need for a `if/else` clause, but it's perhaps a bit too cryptic :)

Reply:
My first encounter with the caret symbol on Python, I had to look it up! The XOR makes it certainly more elegant, but I agree that maybe it can be a little bit confusing too. It might not be as straight forward as the other way in case someone is just quickly inspecting the file, even if the reader is familiar with the symbol. But maybe it's worth changing it anyway, I'm OK with that!